src/TestSuite/IntegrationTestCase.php

```diff
@@ -445,6 +454,11 @@ protected function _assertStatus($min, $max, $message)
             $this->fail('No response set, cannot assert status code.');
         }
         $status = $this->_response->statusCode();
+
+        if ($this->_exception && ($status < $min || $status > $max)) {
+            throw $this->_exception;
```

Comment:
You were right, and fail() can process the exception just fine, minor different output:
```
1) App\Test\TestCase\Controller\AccountControllerTest::testRegister
exception 'RuntimeException' with message 'The "Passwordable" alias has already ...
```

Perfect!